test/test.py

```diff
@@ -876,6 +878,20 @@ def test_multiple_requests_with_illegal_traceparent(self):
 		self.assertFalse('00000000000000000000000000000000' in trace_ids)
 		self.assertEqual(len(parent_ids), 3)
 
+@unittest.skipIf(SPEC_LEVEL < 2, "Trace Context Level 2")
+class TraceContext2Test(TestBase):
+	def test_propagates_random_flag(self):
+		'''
+		harness sends a request with a traceparent with the random flag set
+		expects a valid traceparent from the output header, with the random flag set
+		'''
+		traceparent, tracestate = self.make_single_request_and_get_tracecontext([
+			['traceparent', '00-12345678901234567890123456789012-1234567890123456-02'],
+		])
+		self.assertEqual(traceparent.trace_id.hex(), '12345678901234567890123456789012')
+		self.assertNotEqual(traceparent.parent_id.hex(), '1234567890123456')
+		self.assertTrue(traceparent.trace_flags & (1 << 1) == (1 << 1))
```

Comment:
_nitpick:_ slightly hard to read IMO. Maybe like this?
 
```suggestion
		self.assertEqual(traceparent.trace_flags & (1 << 1), 1 << 1)
```

Or this:

```suggestion
		self.assertEqual(traceparent.trace_flags & 2, 2)
```

Reply:
What about this:

```patch
diff --git a/test/test.py b/test/test.py
index 97d9991..1f8d93f 100644
--- a/test/test.py
+++ b/test/test.py
@@ -114,6 +114,12 @@ class TestBase(unittest.TestCase):
 		headers = self.make_request(headers)[0]['headers']
 		return (self.get_traceparent(headers), self.get_tracestate(headers))
 
+	def assert_bit_set(self, value, n):
+		'''
+		assert that the nth position bit is set where n == 0 is the least significant bit
+		'''
+		self.assertEqual(value & (1 << n - 1), (1 << n - 1))
+
 class TraceContextTest(TestBase):
 	def test_both_traceparent_and_tracestate_missing(self):
 		'''
@@ -890,7 +896,7 @@ class TraceContext2Test(TestBase):
 		])
 		self.assertEqual(traceparent.trace_id.hex(), '12345678901234567890123456789012')
 		self.assertNotEqual(traceparent.parent_id.hex(), '1234567890123456')
-		self.assertTrue(traceparent.trace_flags & (1 << 1) == (1 << 1))
+		self.assert_bit_set(traceparent.trace_flags, 2)
 
 if __name__ == '__main__':
 	if len(sys.argv) >= 2:
```

Reply:
That's splendid 👍